File: Frontend/barbershop/src/types/ResponseDTOs/userVoucherDTO.ts
import type { UserDTO } from "./userDTO";
import type { VoucherDTO } from "./voucherDTO";

export type UserVoucherDTO = {
    UserVoucherId?: number;
    UserId?: number;
    VoucherId?: number;
    Quantity?: number;
    IsActive?: boolean;
    User?: UserDTO;
    Voucher?: VoucherDTO;
};

File: Backend/barbershop/barbershop/Models/ResponseDTOs/UserVoucherDTO.cs
using barbershop.Models.Entitys;

namespace barbershop.Models.ResponseDTOs
{
    public class UserVoucherDTO
    {
        public long UserVoucherId { get; set; }

        public long? UserId { get; set; }

        public int? VoucherId { get; set; }

        public int? Quantity { get; set; }

        public bool? IsActive { get; set; }

        public UserDTO? UserDTO { get; set; }

        public VoucherDTO? VoucherDTO { get; set; }
    }
}
